src/include/mpir_process.h

```diff
@@ -16,6 +16,8 @@
 #include "mpir_netloc.h"
 #endif
 
+#include "mpir_hw_topo.h"
+
```

Comment:
Squash last 3 commits?

Reply:
Also I think a better place to include these headers is `<mpiimpl.h>`.

Reply:
Sorry I should have said this. Please consider only the last 8 commits